fpm/test/cli_test/cli_test.f90

```diff
@@ -0,0 +1,232 @@
+      if(cstat.eq.act_cstat.and.estat.eq.act_estat)then
+          if(estat.eq.0)then
+             open(file='_test_cli',newunit=lun,delim='quote')
+             act_name=[(repeat(' ',len(act_name)),i=1,max_names)]
+             act_release=.false.
+             act_w_e=.false.
+             act_w_t=.false.
+             act_args=repeat(' ',132)
+             read(lun,nml=act_cli,iostat=ios,iomsg=message)
+             if(ios.ne.0)then
+                write(*,'(a)')'ERROR:',trim(message)
+             endif
+             close(unit=lun)
+             ! compare results to expected values
+             subtally=[logical ::]
+             call test_test('NAME',all(act_name.eq.name))
+             call test_test('RELEASE',act_release.eqv.release)
+             call test_test('WITH_EXPECTED',act_w_e.eqv.w_e)
+             call test_test('WITH_TESTED',act_w_t.eqv.w_t)
+             call test_test('WITH_TEST',act_w_t.eqv.w_t)
+             call test_test('ARGS',act_args.eq.args)
+             if(all(subtally))then
+                write(*,'(*(g0))')'PASSED: TEST ',i,' STATUS: expected ',cstat,' ',estat,' actual ',act_cstat,' ',act_estat,&
+                & ' for [',trim(cmd),']'
+                tally=[tally,.true.]
+             else
+                write(*,'(*(g0))')'FAILED: TEST ',i,' STATUS: expected ',cstat,' ',estat,' actual ',act_cstat,' ',act_estat,&
+                & ' for [',trim(cmd),']'
+                print '(4a)', &
+                   'This file was compiled by ', &
+                   compiler_version(),           &
+                   ' using the options ',        &
+                   compiler_options()
+                write(*,nml=act_cli,delim='quote')
+                tally=[tally,.false.]
+             endif
+          else
+             write(*,'(*(g0))')'PASSED: TEST ',i,' EXPECTED BAD STATUS: expected ',cstat,' ',estat, &
+             ' actual ',act_cstat,' ',act_estat,' for [',trim(cmd),']'
+             tally=[tally,.true.]
+          endif
+      else
+          write(*,'(*(g0))')'FAILED: TEST ',i,'BAD STATUS: expected ',cstat,' ',estat,' actual ',act_cstat,' ',act_estat,&
+          ' for [',trim(cmd),']'
+          tally=[tally,.false.]
+      endif
+   enddo
+   ! write up total results and if anything failed exit with a non-zero status
+   write(*,'(*(g0))')'TALLY;',tally
+   if(all(tally))then
+      write(*,'(*(g0))')'PASSED: all ',count(tally),' tests passed '
+   else
+      write(*,*)'FAILED: PASSED=',count(tally),' FAILED=',count(.not.tally)
+      stop 4
+   endif
+else
+   ! call this program with arguments
+   !=============================================
+   debugit: block
+   integer :: j, ilen
+   character(len=256) :: big_argument
+   write(*,*)'arguments seen directly by program'
+   do j=1,command_argument_count()
+      call get_command_argument(number=j,value=big_argument,length=ilen)
+      write(*,'(*(g0))')i,'[',big_argument(:ilen),']'
+   enddo
```

Comment:
`i` is used uninitialized here, while the counter `j` is not used.

Reply:
```suggestion
   do j=1,command_argument_count()
      call get_command_argument(number=j, value=big_argument, length=ilen)
      write(*,'(*(g0))') j, '[', big_argument(:ilen), ']'
   enddo
```